PubSub dependencies are broken

Introduced by #228. That PR added gRPC dependencies that are probably conflicting with PubSub. The failure only happens when trying to authenticate to PubSub, so the tests using the emulator still pass.

```
08:58:16.151 [main] ERROR com.spotify.ffwd.FastForwardAgent - Uncaught exception in thread main, exiting (status = 2)
java.lang.RuntimeException: Failed during initialization
	at com.spotify.ffwd.AgentCore.<init>(AgentCore.java:115)
	at com.spotify.ffwd.AgentCore.<init>(AgentCore.java:91)
	at com.spotify.ffwd.AgentCore$Builder.build(AgentCore.java:467)
	at com.spotify.ffwd.FastForwardAgent.setup(FastForwardAgent.java:106)
	at com.spotify.ffwd.FastForwardAgent.main(FastForwardAgent.java:54)
Caused by: com.google.inject.CreationException: Unable to create injector, see the following errors:

1) Error in custom provider, java.lang.IncompatibleClassChangeError: class io.grpc.auth.GoogleAuthLibraryCallCredentials can not implement io.grpc.CallCredentials, because it is not an interface (io.grpc.CallCredentials is in unnamed module of loader 'app')
  at com.spotify.ffwd.pubsub.PubsubOutputPlugin$1.publisher(PubsubOutputPlugin.java:132) (via modules: com.spotify.ffwd.output.OutputManagerModule$1 -> com.spotify.ffwd.pubsub.PubsubOutputPlugin$1)
  while locating com.google.cloud.pubsub.v1.Publisher
    for field at com.spotify.ffwd.pubsub.PubsubPluginSink.publisher(PubsubPluginSink.java:59)
  while locating com.spotify.ffwd.pubsub.PubsubPluginSink
  at com.spotify.ffwd.pubsub.PubsubOutputPlugin$1.configure(PubsubOutputPlugin.java:189) (via modules: com.spotify.ffwd.output.OutputManagerModule$1 -> com.spotify.ffwd.pubsub.PubsubOutputPlugin$1)
  while locating com.spotify.ffwd.pubsub.PubsubPluginSink annotated with @com.google.inject.name.Named(value="pubsubSink")
  while locating com.spotify.ffwd.output.BatchablePluginSink annotated with @com.spotify.ffwd.output.BatchingDelegate()
    for field at com.spotify.ffwd.output.BatchingPluginSink.sink(BatchingPluginSink.java:56)
  at com.spotify.ffwd.output.OutputPlugin$1.configure(OutputPlugin.java:98) (via modules: com.spotify.ffwd.output.OutputManagerModule$1 -> com.spotify.ffwd.pubsub.PubsubOutputPlugin$1 -> com.spotify.ffwd.output.OutputPlugin$1)

1 error
	at com.google.inject.internal.Errors.throwCreationExceptionIfErrorsExist(Errors.java:543)
	at com.google.inject.internal.InternalInjectorCreator.injectDynamically(InternalInjectorCreator.java:178)
	at com.google.inject.internal.InternalInjectorCreator.build(InternalInjectorCreator.java:109)
	at com.google.inject.internal.InjectorImpl.createChildInjector(InjectorImpl.java:237)
	at com.spotify.ffwd.AgentCore.setupPrimaryInjector(AgentCore.java:374)
	at com.spotify.ffwd.AgentCore.<init>(AgentCore.java:113)
	... 4 more
Caused by: java.lang.IncompatibleClassChangeError: class io.grpc.auth.GoogleAuthLibraryCallCredentials can not implement io.grpc.CallCredentials, because it is not an interface (io.grpc.CallCredentials is in unnamed module of loader 'app')
	at java.base/java.lang.ClassLoader.defineClass1(Native Method)
	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1017)
	at java.base/java.security.SecureClassLoader.defineClass(SecureClassLoader.java:174)
	at java.base/jdk.internal.loader.BuiltinClassLoader.defineClass(BuiltinClassLoader.java:800)
	at java.base/jdk.internal.loader.BuiltinClassLoader.findClassOnClassPathOrNull(BuiltinClassLoader.java:698)
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClassOrNull(BuiltinClassLoader.java:621)
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:579)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	at com.google.api.gax.grpc.GrpcCallContext.withCredentials(GrpcCallContext.java:126)
	at com.google.api.gax.grpc.GrpcCallContext.withCredentials(GrpcCallContext.java:64)
	at com.google.api.gax.rpc.ClientContext.create(ClientContext.java:159)
	at com.google.cloud.pubsub.v1.stub.GrpcPublisherStub.create(GrpcPublisherStub.java:161)
	at com.google.cloud.pubsub.v1.Publisher.<init>(Publisher.java:154)
	at com.google.cloud.pubsub.v1.Publisher.<init>(Publisher.java:83)
	at com.google.cloud.pubsub.v1.Publisher$Builder.build(Publisher.java:607)
	at com.spotify.ffwd.pubsub.PubsubOutputPlugin$1.publisher(PubsubOutputPlugin.java:157)
	at com.spotify.ffwd.pubsub.PubsubOutputPlugin$1$$FastClassByGuice$$5c879ecf.invoke(<generated>)
	at com.google.inject.internal.ProviderMethod$FastClassProviderMethod.doProvision(ProviderMethod.java:264)
	at com.google.inject.internal.ProviderMethod.doProvision(ProviderMethod.java:173)
	at com.google.inject.internal.InternalProviderInstanceBindingImpl$CyclicFactory.provision(InternalProviderInstanceBindingImpl.java:185)
	at com.google.inject.internal.InternalProviderInstanceBindingImpl$CyclicFactory.get(InternalProviderInstanceBindingImpl.java:162)
	at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
	at com.google.inject.internal.SingletonScope$1.get(SingletonScope.java:168)
	at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:39)
	at com.google.inject.internal.SingleFieldInjector.inject(SingleFieldInjector.java:52)
	at com.google.inject.internal.MembersInjectorImpl.injectMembers(MembersInjectorImpl.java:147)
	at com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:124)
	at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:91)
	at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:306)
	at com.google.inject.internal.FactoryProxy.get(FactoryProxy.java:62)
	at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
	at com.google.inject.internal.SingletonScope$1.get(SingletonScope.java:168)
	at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:39)
	at com.google.inject.internal.FactoryProxy.get(FactoryProxy.java:62)
	at com.google.inject.internal.SingleFieldInjector.inject(SingleFieldInjector.java:52)
	at com.google.inject.internal.MembersInjectorImpl.injectMembers(MembersInjectorImpl.java:147)
	at com.google.inject.internal.MembersInjectorImpl.injectAndNotify(MembersInjectorImpl.java:101)
	at com.google.inject.internal.Initializer$InjectableReference.get(Initializer.java:245)
	at com.google.inject.internal.Initializer.injectAll(Initializer.java:140)
	at com.google.inject.internal.InternalInjectorCreator.injectDynamically(InternalInjectorCreator.java:176)
	... 8 more```

--- a/api/pom.xml
+++ b/api/pom.xml
@@ -70,6 +70,10 @@
       <artifactId>jackson-databind</artifactId>
     </dependency>
     <dependency>
+      <groupId>com.fasterxml.jackson.core</groupId>
+      <artifactId>jackson-core</artifactId>
+    </dependency>
+    <dependency>
       <groupId>com.fasterxml.jackson.datatype</groupId>
       <artifactId>jackson-datatype-jdk8</artifactId>
     </dependency>

--- a/core/pom.xml
+++ b/core/pom.xml
@@ -47,12 +47,6 @@
     <dependency>
       <groupId>com.google.protobuf</groupId>
       <artifactId>protobuf-java</artifactId>
-      <exclusions>
-        <exclusion>
-          <groupId>com.google.guava</groupId>
-          <artifactId>guava</artifactId>
-        </exclusion>
-      </exclusions>
     </dependency>
 
     <dependency>

--- a/modules/carbon/pom.xml
+++ b/modules/carbon/pom.xml
@@ -28,7 +28,6 @@
     <dependency>
       <groupId>org.apache.commons</groupId>
       <artifactId>commons-lang3</artifactId>
-      <version>3.3.2</version>
     </dependency>
   </dependencies>
 </project>

--- a/modules/opencensus/pom.xml
+++ b/modules/opencensus/pom.xml
@@ -45,12 +45,10 @@
     <dependency>
       <groupId>io.opencensus</groupId>
       <artifactId>opencensus-api</artifactId>
-      <version>0.23.0</version>
     </dependency>
     <dependency>
       <groupId>io.opencensus</groupId>
       <artifactId>opencensus-impl</artifactId>
-      <version>0.23.0</version>
       <exclusions>
         <exclusion>
           <groupId>com.google.guava</groupId>
@@ -99,7 +97,15 @@
           <groupId>com.google.guava</groupId>
           <artifactId>guava</artifactId>
         </exclusion>
+        <exclusion>
+          <groupId>com.fasterxml.jackson.core</groupId>
+          <artifactId>jackson-core</artifactId>
+        </exclusion>
       </exclusions>
+    </dependency>
+    <dependency>
+      <groupId>com.fasterxml.jackson.core</groupId>
+      <artifactId>jackson-core</artifactId>
     </dependency>
     <dependency>
       <groupId>com.google.code.gson</groupId>

--- a/modules/pubsub/pom.xml
+++ b/modules/pubsub/pom.xml
@@ -31,29 +31,21 @@
         </dependency>
 
         <dependency>
+            <groupId>com.fasterxml.jackson.core</groupId>
+            <artifactId>jackson-core</artifactId>
+        </dependency>
+        <dependency>
             <groupId>com.google.cloud</groupId>
             <artifactId>google-cloud-pubsub</artifactId>
-            <version>1.36.0</version>
+            <version>1.110.3</version>
             <exclusions>
                 <exclusion>
                     <groupId>com.google.guava</groupId>
                     <artifactId>guava</artifactId>
                 </exclusion>
                 <exclusion>
-                    <groupId>com.google.guava</groupId>
-                    <artifactId>guava-jdk5</artifactId>
-                </exclusion>
-                <exclusion>
-                    <groupId>com.google.auto.value</groupId>
-                    <artifactId>auto-value</artifactId>
-                </exclusion>
-                <exclusion>
-                    <groupId>com.google.protobuf</groupId>
-                    <artifactId>protobuf-java</artifactId>
-                </exclusion>
-                <exclusion>
-                    <groupId>com.google.api.grpc</groupId>
-                    <artifactId>proto-google-common-protos</artifactId>
+                    <groupId>com.fasterxml.jackson.core</groupId>
+                    <artifactId>jackson-core</artifactId>
                 </exclusion>
             </exclusions>
         </dependency>
@@ -62,31 +54,9 @@
             <artifactId>protobuf-java</artifactId>
         </dependency>
         <dependency>
-            <groupId>com.google.api.grpc</groupId>
-            <artifactId>proto-google-common-protos</artifactId>
-        </dependency>
-        <dependency>
             <groupId>com.google.guava</groupId>
             <artifactId>guava</artifactId>
             <version>27.0.1-jre</version>
-            <exclusions>
-                <exclusion>
-                    <groupId>com.google.errorprone</groupId>
-                    <artifactId>error_prone_annotations</artifactId>
-                </exclusion>
-                <exclusion>
-                    <groupId>org.codehaus.mojo</groupId>
-                    <artifactId>animal-sniffer-annotations</artifactId>
-                </exclusion>
-            </exclusions>
-        </dependency>
-        <dependency>
-            <groupId>com.google.errorprone</groupId>
-            <artifactId>error_prone_annotations</artifactId>
-        </dependency>
-        <dependency>
-            <groupId>org.codehaus.mojo</groupId>
-            <artifactId>animal-sniffer-annotations</artifactId>
         </dependency>
 
         <!-- testing -->

--- a/pom.xml
+++ b/pom.xml
@@ -64,16 +64,17 @@
     <tiny-async.version>1.11.0</tiny-async.version>
     <log4j.version>2.13.3</log4j.version>
     <slf4j.version>1.7.25</slf4j.version>
-    <jackson.version>2.9.9</jackson.version>
-    <kotlin.version>1.3.31</kotlin.version>
+    <jackson.version>2.10.0</jackson.version>
+    <kotlin.version>1.3.41</kotlin.version>
     <spotify-metrics.version>1.1.2</spotify-metrics.version>
     <findbugs.version>3.0.4</findbugs.version>
     <grpc.version>1.35.0</grpc.version>
+    <protobuf.version>3.14.0</protobuf.version>
     <gson.version>2.8.6</gson.version>
     <errorprone.version>2.4.0</errorprone.version>
     <animal_sniffer.version>1.19</animal_sniffer.version>
     <guava.version>27.0.1-jre</guava.version>
-    <protobuf.version>3.11.1</protobuf.version>
+    <opencensus.version>0.28.0</opencensus.version>
   </properties>
 
   <profiles>
@@ -213,21 +214,6 @@
       </dependency>
 
       <dependency>
-        <groupId>com.google.auth</groupId>
-        <artifactId>google-auth-library-credentials</artifactId>
-        <version>0.17.1</version>
-      </dependency>
-      <dependency>
-        <groupId>com.google.auth</groupId>
-        <artifactId>google-auth-library-oauth2-http</artifactId>
-        <version>0.17.1</version>
-      </dependency>
-      <dependency>
-        <groupId>com.google.http-client</groupId>
-        <artifactId>google-http-client</artifactId>
-        <version>1.31.0</version>
-      </dependency>
-      <dependency>
         <groupId>com.google.j2objc</groupId>
         <artifactId>j2objc-annotations</artifactId>
         <version>1.3</version>
@@ -236,11 +222,6 @@
         <groupId>com.google.code.gson</groupId>
         <artifactId>gson</artifactId>
         <version>${gson.version}</version>
-      </dependency>
-      <dependency>
-        <groupId>com.google.protobuf</groupId>
-        <artifactId>protobuf-java</artifactId>
-        <version>${protobuf.version}</version>
       </dependency>
 
       <!-- grpc dependencies -->
@@ -253,12 +234,6 @@
         <groupId>io.grpc</groupId>
         <artifactId>grpc-core</artifactId>
         <version>${grpc.version}</version>
-        <exclusions>
-          <exclusion>
-            <groupId>com.google.protobuf</groupId>
-            <artifactId>protobuf-java</artifactId>
-          </exclusion>
-        </exclusions>
       </dependency>
       <dependency>
         <groupId>io.grpc</groupId>
@@ -281,11 +256,6 @@
         <version>6.0.53</version>
       </dependency>
       <dependency>
-        <groupId>com.google.api.grpc</groupId>
-        <artifactId>proto-google-common-protos</artifactId>
-        <version>1.12.0</version>
-      </dependency>
-      <dependency>
         <groupId>com.google.errorprone</groupId>
         <artifactId>error_prone_annotations</artifactId>
         <version>${errorprone.version}</version>
@@ -295,10 +265,27 @@
         <artifactId>animal-sniffer-annotations</artifactId>
         <version>${animal_sniffer.version}</version>
       </dependency>
+      <dependency>
+        <groupId>com.google.protobuf</groupId>
+        <artifactId>protobuf-java</artifactId>
+        <version>${protobuf.version}</version>
+      </dependency>
+      <dependency>
+        <groupId>com.google.cloud</groupId>
+        <artifactId>libraries-bom</artifactId>
+        <version>16.3.0</version>
+        <type>pom</type>
+        <scope>import</scope>
+      </dependency>
 
       <dependency>
         <groupId>org.jetbrains.kotlin</groupId>
         <artifactId>kotlin-stdlib-jdk8</artifactId>
+        <version>${kotlin.version}</version>
+      </dependency>
+      <dependency>
+        <groupId>org.jetbrains.kotlin</groupId>
+        <artifactId>kotlin-reflect</artifactId>
         <version>${kotlin.version}</version>
       </dependency>
       <dependency>
@@ -313,6 +300,16 @@
         </exclusions>
       </dependency>
 
+      <dependency>
+        <groupId>io.opencensus</groupId>
+        <artifactId>opencensus-api</artifactId>
+        <version>${opencensus.version}</version>
+      </dependency>
+      <dependency>
+        <groupId>io.opencensus</groupId>
+        <artifactId>opencensus-impl</artifactId>
+        <version>${opencensus.version}</version>
+      </dependency>
 
       <dependency>
         <groupId>com.spotify.metrics</groupId>
@@ -332,12 +329,6 @@
         <version>${jackson.version}</version>
         <scope>import</scope>
         <type>pom</type>
-      </dependency>
-      <!-- annotations in the bom don't use patch releases which causes a conflict -->
-      <dependency>
-        <groupId>com.fasterxml.jackson.core</groupId>
-        <artifactId>jackson-annotations</artifactId>
-        <version>${jackson.version}</version>
       </dependency>
 
       <dependency>
@@ -373,6 +364,11 @@
         <groupId>org.isomorphism</groupId>
         <artifactId>token-bucket</artifactId>
         <version>1.6</version>
+      </dependency>
+      <dependency>
+        <groupId>org.apache.commons</groupId>
+        <artifactId>commons-lang3</artifactId>
+        <version>3.5</version>
       </dependency>
 
       <dependency>

